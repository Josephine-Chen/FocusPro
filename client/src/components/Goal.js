import React from 'react';
import { Input, Button } from 'react-materialize';
import Nouislider from 'react-nouislider';
import Subgoal from './Subgoal';

// Todo: Replace with better slider
export default class Goal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      subgoals: [],
      subgoal: '',
      step: 1,
    };
    this.handleChange = this.handleChange.bind(this);
    this.postSubgoal = this.postSubgoal.bind(this);
    this.getSubgoals = this.getSubgoals.bind(this);
    this.getStep = this.getStep.bind(this);
    this.handleKeyPress = this.handleKeyPress.bind(this);
  }

  componentWillMount() {
    this.getSubgoals();
  }

  componentDidMount() {
    this.getSubgoals();
    this.callCustomJQuery();
  }

  getSubgoals() {
    const that = this;
    $.ajax({
      type: 'GET', // GET REQUEST
      url: `/api/goals/${this.props.goal}/subgoals`,
      success: (data) => {
        console.log('SUCCESS: OBTAINED ALL SUBGOALS:', JSON.stringify(data.data));
        that.setState({ subgoals: data.data });
        that.getStep();
      },
      error: (err) => { console.log('ERROR: COULD NOT GET SUBGOALS', err); },
    });
  }

  getStep() {
    if (this.state.subgoals.length > 0) {
      const newStep = Math.round(100 / this.state.subgoals.length);
      this.setState({ step: newStep });
    } else {
      this.setState({ step: 1 });
    }
  }

  handleChange(event) {
    this.setState({ subgoal: event.target.value });
    this.getStep();
  }

  callCustomJQuery() {
    $('.collapsible').collapsible();
  }

  postSubgoal() {
    const that = this;
    $.ajax({
      type: 'POST',
      url: `/api/goals/${this.props.goal}/subgoals`,
      contentType: 'application/json',
      data: JSON.stringify({ subgoal: this.state.subgoal, status: false, GoalId: this.props.goal }),
      success: (data) => {
        console.log('SUCCESS: POSTED INDIVIDUAL GOAL: ', data);
        that.getSubgoals();
        that.cleanInput();
        that.alertConfirmation();
      },
      error: (err) => { console.log('ERROR: COULD NOT POST INDIVIDUAL GOAL', err); },
    });
  }

  alertConfirmation() {
    Materialize.toast('Subgoal set!', 1000);
  }

  cleanInput() {
    this.setState({ subgoal: '' });
  }

  handleKeyPress(e){
    if(e.key == 'Enter'){
      this.postSubgoal();
    }
  }

  render() {
    return (
      <div className="collapsible-body">Progress:
        <br /><br /><br />
        <Nouislider
          range={{ min: 0, max: 100 }}
          start={[0]}
          step={this.state.step}
          behaviour={'tap'}
          format={{
            from: value => `${parseInt(value, 10)}%`,
            to: value => `${parseInt(value, 10)}%`,
          }}
          tooltips
        />
        <br />
        Subgoals:
        <br />
        {this.state.subgoals.length === 0 && <div> There is no subgoal set currently. </div>}
        {this.state.subgoals.map(subgoal =>
          <Subgoal key={`sub ${subgoal.id}`} subgoal={subgoal} status={subgoal.status} id={subgoal.id} user_id={this.props.user_id} goal={subgoal.GoalId} updateSubgoals={this.getSubgoals} />
          )}
        <Input s={8} label="New Subgoal" onChange={this.handleChange} value={this.state.subgoal} onKeyPress={this.handleKeyPress}/>
        <Button className="subgoalButton" waves="light" onClick={this.postSubgoal}>Set Subgoal</Button>
      </div>
    );
  }
}
