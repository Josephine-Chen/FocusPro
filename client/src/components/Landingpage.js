import React from 'react';
import { Row, Col, Card } from 'react-materialize';

// Todo: When using actual images, set max-height and max-width based on % so it scales

export default class Landingpage extends React.Component {
  render() {
    return (
      <div className="row">
        <h3 className="center"> Welcome to FocusPro.</h3>
        <h5 className="center"> We help you limit distractions and stay on task. </h5>
        <br /> <br />
        <Row>
          <Col s={7}>
            <img className="placeholder" src="../../assets/Landingpage-goals.jpeg" alt="goals" />
          </Col>
          <Col s={5}>
            <br /> <br /><br /> <br /> <br />
            FocusPro helps you set goals while tracking your progress in accomplishing them.
            You can blacklist websites to limit distractions by either blocking them directly, or having FocusPro warn you when you have exceeded the time limit you have set.
          </Col>
        </Row>
        <br /> <br />
        <Row>
          <Col s={7}>
            <br /> <br /><br /> <br /> <br />
            Alerts you when you are off task through window notifications using our Chrome extension.
            Shows prompts to ask you to self-reflect on your progress for the day and how much you have worked towards your goal.
          </Col>
          <Col s={5}>
            <img className="placeholder" alt="Chrome extension notification" />
          </Col>
        </Row>
        <br /> <br />
        <Row>
          <Col s={7}>
            <img className="placeholder" src="../../assets/Landingpage-stats.jpeg" alt="stats for site" />
          </Col>
          <Col s={5}>
            <br /> <br /><br /> <br /> <br />
            Intelligently keeps track of overall data patterns.
            See how your habits change as you slowly wean yourself off unproductive sites.
          </Col>
        </Row>
        <br /> <br />
        <Row>
          <Col s={7}>
            <br /> <br /><br /> <br /> <br />
            <h5>Companion Google Chrome Extension</h5>
            Use our Chrome extension to track your tab usage and to send you alert messages and self-reflection questions.
            Download here.
          </Col>
          <Col s={5}>
            <img src="../../assets/icon128.png" alt="Chrome extension icon" />
          </Col>
        </Row>
        <br />
        <div className="center">
          <a className="waves-effect waves-light btn-large">Get Started</a>
        </div>
        <br />
        <h5 className="center"> Meet the Team </h5>
        <Row>
          <Col>
            <Card key={10} className="white" title="Jeff Bentler" actions={[<a href="https://github.com/bentljm">GitHub</a>]}>
              <img src="../../assets/jeff_img.jpeg" className="circle" alt="Jeff" />
            </Card>
          </Col>
          <Col>
            <Card key={11} className="white" title="Josephine Chen" actions={[<a href="https://github.com/Josephine-Chen">GitHub</a>]}>
              <img src="../../assets/josephine_img.jpeg" className="circle" alt="Josephine" />
            </Card>
          </Col>
          <Col>
            <Card key={12} className="white" title="Ian George" actions={[<a href="https://github.com/OppenheimerAndTheMartians">GitHub</a>]}>
              <img src="../../assets/ian_img.jpeg" className="circle" alt="Ian" />
            </Card>
          </Col>
          <Col>
            <Card key={13} className="white" title="Sherry Hsu" actions={[<a href="https://github.com/SherryH">GitHub</a>]}>
              <img src="../../assets/sherry_img.jpeg" className="circle" alt="Sherry" />
            </Card>
          </Col>
        </Row>
      </div>
    );
  }
}
