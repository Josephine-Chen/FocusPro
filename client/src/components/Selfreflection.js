import React from 'react';
import { BootstrapTable, TableHeaderColumn } from 'react-bootstrap-table';
import moment from 'moment';
import { getReflectionsAjax } from '../utils/SettingsUtil';

export default class Selfreflection extends React.Component {
  constructor(props) { // Hand downs the prop chain
    super(props);
    this.state = {
      profile: this.props.auth.getProfile(), // it gets imported here
      reflections: [],
      answer: '',
      question: '',
    };
    this.getReflections = this.getReflections.bind(this);
    this.handleCAnswerhange = this.handleAnswerChange.bind(this);

    this.options = {
      defaultSortName: 'date',  // default sort column
      defaultSortOrder: 'asc',  // default sort order
    };
  }

  componentDidMount() {
    this.getReflections();
    this.callCustomJQuery();
  }

  getReflections() {
    const that = this;
    getReflectionsAjax(this.state.profile.user_id, (data) => {
      const reflectionData = data.data.slice();
      // transform data into format accepted by datatable
      (data.data).forEach((refl, index) => {
        reflectionData[index].date = moment(refl.updatedAt).calendar();
      });
      that.setState({
        reflections: reflectionData,
      });
    });
  }

  getCaret(direction) {
    if (direction === 'asc') {
      return ( <i className="fa fa-caret-up" aria-hidden="true" /> );
    }
    if (direction === 'desc') {
      return (
        <i className="fa fa-caret-down" aria-hidden="true" />
      );
    }
    return (
      <i className="fa fa-sort" aria-hidden="true" />
    );
  }

  handleAnswerChange(event) {
    this.setState({
      answer: event.target.value
    });
  }

  callCustomJQuery() {
    $('.collapsible').collapsible();
  }

  render() {
    return (
      <div>
        <h1> Self-Reflection </h1>

        <BootstrapTable data={this.state.reflections} options={this.options} striped hover pagination>
          <TableHeaderColumn hidden isKey dataField="id">ID</TableHeaderColumn>
          <TableHeaderColumn dataSort dataField="question" tdStyle={{ whiteSpace: 'normal' }} caretRender={this.getCaret}>Question </TableHeaderColumn>
          <TableHeaderColumn dataField="answer" tdStyle={{ whiteSpace: 'normal' }}>Answer </TableHeaderColumn>
          <TableHeaderColumn dataSort dataField="date" tdStyle={{ whiteSpace: 'normal' }} caretRender={this.getCaret} width="100">Date </TableHeaderColumn>
        </BootstrapTable>
      </div>
    );
  }
}

