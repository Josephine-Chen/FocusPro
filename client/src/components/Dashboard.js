import React from 'react';
import {Row, Input, Col, Button, Icon} from 'react-materialize';
import Goal from './Goal.js';
import Site from './Site.js';
import Stat from './Stat.js';
import Motivational from './Motivational.js';

export default class Dashboard extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      profile: this.props.auth.getProfile(),
      goals: [],
      goalInput: '',
      dayGoalInput: '',
      userId: '',
      setting: {},
      blacklist: []
    };
    this.handleInputChange = this.handleInputChange.bind(this);
    this.postGoal = this.postGoal.bind(this);
    this.handleDayGoalChange = this.handleDayGoalChange.bind(this);
    this.handleDayGoalSubmission = this.handleDayGoalSubmission.bind(this);
    this.removeGoal = this.removeGoal.bind(this);
  }

  componentWillMount() {
    this.getUserId();
    this.getSetting();
  }

  componentDidMount(){
    this.getAllGoals();
    this.getBlacklist();
  }

  getUserId() {
    var that = this;
    $.ajax({
      type: 'GET',
      url: 'api/users/' + this.state.profile.user_id,
      success: function (data) {
        console.log("SUCCESS: GOT USER INFO", data.data[0]);
        that.setState({userId: data.data[0].id});
        that.setState({dayGoalInput: data.data[0].daily_goal || ''});
      },
      error: function (err) {
        console.log('ERROR: COULD NOT GET USERID', err);
      }
    });
  }

  getSetting() {
    var that = this;
    $.ajax({
      type: 'GET', // GET REQUEST
      url: '/api/users/' + this.state.profile.user_id + '/setting',
      success: function(data) {
        console.log("SUCCESS: OBTAINED SETTINGS: ", data);
        that.setState({setting: data.data});
      },
      error: function(err) {console.log("ERROR: COULD NOT GET SETTINGS", err);}
    });
  }

  getBlacklist() {
    var that = this;
    $.ajax({
      type: 'GET', // GET REQUEST
      url: '/api/users/' + this.state.profile.user_id + '/setting/blacklist',
      success: function(data) {
        console.log("SUCCESS: OBTAINED BLACKLIST: ", data.data);
        that.setState({blacklist: data.data});
      },
      error: function(err) {console.log("ERROR: COULD NOT GET BLACKLIST", err);}
    });
  }

  getAllGoals () {
    var that = this;
    $.ajax({
      type: 'GET', // GET REQUEST
      url: '/api/users/' + this.state.profile.user_id + '/goals/',
      success: function(data) {
        console.log("SUCCESS: OBTAINED ALL GOALS:", data);
        that.setState({goals: data.data});
      },
      error: function(err) {
        console.log("ERROR: COULD NOT GET ALL GOALS", err);
      }
    });
  }

  handleInputChange(event) {
    this.setState({goalInput: event.target.value});
  }

  handleDayGoalChange(event) {
    this.setState({dayGoalInput: event.target.value});
  }

  handleDayGoalSubmission() {
    $.ajax({
      type: 'PUT',
      url: '/api/users/' + this.state.profile.user_id,
      contentType: 'application/json',
      data: JSON.stringify({daily_goal: this.state.dayGoalInput}),
      success: function(data) {console.log("Update daily goal to", data);},
      error: function(err) {console.log("Error updating daily goal", err);}
    });
  }

  postGoal() {
    var that = this;
    $.ajax({
      type: 'POST',
      url: '/api/users/' + this.state.profile.user_id + '/goals',
      contentType: 'application/json',
      data: JSON.stringify({goal: this.state.goalInput, progress: 0, goal_picture: '', UserId: this.state.userId}),
      success: function(data) {
        console.log("SUCCESS: POSTED INDIVIDUAL GOAL: ", data);
        that.setState({goalInput: ''}); //Not clearing input...
        that.getAllGoals();
      },
      error: function(err) {console.log("ERROR: COULD NOT POST INDIVIDUAL GOAL", err);}
    });

  }

  removeGoal (goal_id) {
    var that = this;
    $.ajax({
      type: 'DELETE',
      url: '/api/users/' + this.state.profile.user_id + '/goals/' + goal_id,
      success: function(data) {
        console.log("Remove goal:", data);
        that.getAllGoals();
      },
      error: function(err) {
        console.log("ERROR: COULD NOT GET ALL GOALS", err);
      }
    });
  }

  render() {
    return (
      <div>
        <h1> Welcome, {this.state.profile.given_name} </h1>
        {this.state.setting.quote}
        <br />
        <br />
        <h3> Goal of the Day: </h3>
        <Row>
          <Input s={10} value={this.state.dayGoalInput} onChange={this.handleDayGoalChange} /> <Button className="dayGoalButton" waves='light' onClick={this.handleDayGoalSubmission}>Save</Button>
        </Row>
        <h3> Main Goals: </h3>
          <ul className="collapsible" data-collapsible="expandable">
          {this.state.goals.map((goal, index) => (
            <li key = {index}>
            <div className="collapsible-header">{goal.goal} <a href = '#/dashboard' onClick = {()=>this.removeGoal(goal.id)}><Icon right>delete</Icon></a></div>
            <Goal key = {'goal' + index} goal = {goal.id} user_id = {this.state.profile.user_id} />
          </li>
            ))}
        </ul>
        <Row>
        <Input s={8} label="New Goal" value={this.state.value} onChange={this.handleInputChange} /> <Button className="goalButton" waves='light' onClick={this.postGoal}>Set Goal</Button> <Motivational />
        </Row>
        <br />
        <h3> Sites: </h3>
        <ul className="collapsible" data-collapsible="expandable">
          {this.state.blacklist.map((site, index) => (
            <li key = {'sites ' + index}>
              <div className="collapsible-header">{site.url}</div>
              <div className="collapsible-body"><Site url = {site.url} siteId = {site.id}/></div>
            </li>
            ))}
        </ul>
        <br />
        <br />
        <h3> Stats: </h3>
        <Row>
          <Col s={4}>
          <Stat />
          Today's Stats
          </Col>
          <Col s={4}>
          <Stat />
          Week's Stats
          </Col>
          <Col s={4}>
          <Stat />
          All Time Stats
          </Col>
        </Row>
      </div>
    );
  }
}