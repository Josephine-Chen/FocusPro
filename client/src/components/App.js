import React, { Component } from 'react';
import Landingpage from './Landingpage.js';
import {Sidebar} from './Sidebar.js';
import Auth from './Auth.js';

export default class App extends React.Component {

  constructor (props) {
    super(props);
  }

  componentWillMount () {
      this.lock = new Auth0Lock('V3oJOmHRIeW7nJnLZIa7ioWOo8godJVQ', 'focuspro.auth0.com');
      // Set the state with a property that has the token
      this.setState({idToken: this.getIdToken()})
  }

  createLock () {
    this.lock = new Auth0Lock(this.props.clientId, this.props.domain);
  }

  getIdToken () {
    // First, check if there is already a JWT in local storage
    var idToken = localStorage.getItem('id_token');
    var authHash = this.lock.parseHash(window.location.hash);
    // If there is no JWT in local storage and there is one in the URL hash,
    // save it in local storage
    if (!idToken && authHash) {
      if (authHash.id_token) {
        idToken = authHash.id_token
        localStorage.setItem('id_token', authHash.id_token);
      }
      if (authHash.error) {
        // Handle any error conditions
        console.log("Error signing in", authHash);
      }
    }
    return idToken;
  }

  render() {
    return (
      <div>
        <Sidebar />
        <main>
          <div className="container">
            <Auth lock={this.lock} />
            {this.props.children}
          </div>
        </main>
      </div>
    );
  }
}



