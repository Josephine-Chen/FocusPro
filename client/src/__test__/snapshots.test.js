import React from 'react';
import renderer from 'react-test-renderer';
import App from '../components/App.js';
import Landingpage from '../components/Landingpage.js';
import Dashboard from '../components/Dashboard.js';
import Goal from '../components/Goal.js'
import Subgoal from '../components/Subgoal.js'
import Selfreflection from '../components/Selfreflection.js';
import Settings from '../components/Settings.js';

describe('App (Snapshot)', () => {
  it('App renders', () => {
    //remove this test. children not rendered, pointless
    var route = {auth: 'dummy'};
    // const component = renderer.create(<App route={route}/>);
    // const json = component.toJSON();
    // expect(json).toMatchSnapshot();
  });
});

describe('Landingpage (Snapshot)', () => {
  beforeAll(() => {
    const auth = {
      getProfile: ()=>'dummy',
      'event': {
        on : ()=>'dummy event listener'
      }
    };
  });

  it('Landingpage renders', () => {
    const component = renderer.create(<Landingpage />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('Dashboard (Snapshot)', () => {
  beforeAll(() => {
    const auth = {
      getProfile: ()=>'dummy',
      'event': {
        on : ()=>'dummy event listener'
      }
    };
  });

  it('Dashboard renders', () => {
    const component = renderer.create(<Dashboard />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('Goal (Snapshot)', () => {
  beforeAll(() => {
    const auth = {
      getProfile: ()=>'dummy',
      'event': {
        on : ()=>'dummy event listener'
      }
    };
  });

  it('Goal renders', () => {
    const component = renderer.create(<Goal />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('Subgoal (Snapshot)', () => {
  beforeAll(() => {
    const auth = {
      getProfile: ()=>'dummy',
      'event': {
        on : ()=>'dummy event listener'
      }
    };
  });

  it('Subgoal renders', () => {
    const component = renderer.create(<Subgoal />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('Selfreflection (Snapshot)', () => {
  beforeAll(() => {
    const auth = {
      getProfile: ()=>'dummy',
      'event': {
        on : ()=>'dummy event listener'
      }
    };
  });

  it('Selfreflection renders', () => {
    const component = renderer.create(<Selfreflection />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('Settings (Snapshot)', () => {
  beforeAll(() => {
    const auth = {
      getProfile: ()=>'dummy',
      'event': {
        on : ()=>'dummy event listener'
      }
    };
  });

  it('Settings renders', () => {
    const component = renderer.create(<Settings />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});


