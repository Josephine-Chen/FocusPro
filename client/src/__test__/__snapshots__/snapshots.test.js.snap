exports[`Dashboard (Snapshot) Dashboard renders 1`] = `
<div>
  <h1>
     Welcome, USER 
  </h1>
  "The way to get started is to quit talking and begin doing." - Walt Disney
  <br />
  <br />
  <h3>
     Goal of the Day: 
  </h3>
  <div
    className="row">
    <div
      className="col input-field s12">
      <input
        className=""
        defaultValue={undefined}
        id="input_0"
        onChange={[Function]}
        placeholder={undefined}
        type="text" />
    </div>
  </div>
  <h3>
     Main Goals: 
  </h3>
  <ul
    className="collapsible"
    data-collapsible="expandable">
    <li>
      <div
        className="collapsible-header">
        Goal
      </div>
      <div
        className="collapsible-body">
        <span>
          <form
            action="#">
            <p
              className="range-field">
              <input
                id="slider"
                max="100"
                min="0"
                type="range" />
            </p>
          </form>
          <div>
            <div
              className="row">
              <div
                className="col">
                <input
                  className=""
                  defaultValue={undefined}
                  id="input_1"
                  name="group1"
                  onChange={[Function]}
                  placeholder={undefined}
                  type="checkbox"
                  value="check" />
                <label
                  className=""
                  data-error={undefined}
                  data-success={undefined}
                  htmlFor="input_1">
                  Subgoal
                </label>
              </div>
            </div>
          </div>
        </span>
      </div>
    </li>
    <li>
      <div
        className="collapsible-header">
        Second
      </div>
      <div
        className="collapsible-body">
        <span>
          Map for every goal.
        </span>
      </div>
    </li>
    <li>
      <div
        className="collapsible-header">
        Third
      </div>
      <div
        className="collapsible-body">
        <span>
          Map for every goal.
        </span>
      </div>
    </li>
  </ul>
  <div
    className="row">
    <div
      className="col input-field s8">
      <input
        className=""
        defaultValue={undefined}
        id="input_2"
        onChange={[Function]}
        placeholder={undefined}
        type="text" />
      <label
        className=""
        data-error={undefined}
        data-success={undefined}
        htmlFor="input_2">
        New Goal
      </label>
    </div>
     
    <button
      className="btn waves-effect waves-light goalButton"
      disabled={false}
      onClick={[Function]}>
      Set Goal
    </button>
     
    <button
      className="btn waves-effect waves-light goalButton"
      disabled={false}
      onClick={[Function]}>
      Inspire Me
    </button>
  </div>
  <br />
  <h3>
     Sites: 
  </h3>
  <ul
    className="collapsible"
    data-collapsible="expandable">
    <li>
      <div
        className="collapsible-header">
        Site 1
      </div>
      <div
        className="collapsible-body">
        <span>
          <div>
            Image of line chart goes here. Along with whatever text.
          </div>
        </span>
      </div>
    </li>
    <li>
      <div
        className="collapsible-header">
        Site 2
      </div>
      <div
        className="collapsible-body">
        <span>
          <div>
            Image of line chart goes here. Along with whatever text.
          </div>
        </span>
      </div>
    </li>
    <li>
      <div
        className="collapsible-header">
        Site 3
      </div>
      <div
        className="collapsible-body">
        <span>
          <div>
            Image of line chart goes here. Along with whatever text.
          </div>
        </span>
      </div>
    </li>
  </ul>
  <br />
  <br />
  <h3>
     Stats: 
  </h3>
  <div
    className="row">
    <div
      className="col s4">
      <div>
        Placeholder pie chart goes here.
      </div>
      Today's Stats
    </div>
    <div
      className="col s4">
      <div>
        Placeholder pie chart goes here.
      </div>
      Week's Stats
    </div>
    <div
      className="col s4">
      <div>
        Placeholder pie chart goes here.
      </div>
      All Time Stats
    </div>
  </div>
</div>
`;

exports[`Goal (Snapshot) Goal renders 1`] = `
<div
  className="collapsible-body">
  <span>
    <form
      action="#">
      <p
        className="range-field">
        <input
          id="slider"
          max="100"
          min="0"
          type="range" />
      </p>
    </form>
    <div>
      <div
        className="row">
        <div
          className="col">
          <input
            className=""
            defaultValue={undefined}
            id="input_4"
            name="group1"
            onChange={[Function]}
            placeholder={undefined}
            type="checkbox"
            value="check" />
          <label
            className=""
            data-error={undefined}
            data-success={undefined}
            htmlFor="input_4">
            Subgoal
          </label>
        </div>
      </div>
    </div>
  </span>
</div>
`;

exports[`Landingpage (Snapshot) Landingpage renders 1`] = `
<div
  className="row">
  <h1
    className="center">
     FocusPro 
  </h1>
  <h3
    className="center">
     Welcome to FocusPro.
  </h3>
  <h5
    className="center">
     We help you limit distractions and stay on task. 
  </h5>
  <br />
   
  <br />
  <div
    className="row">
    <div
      className="col s7">
      <img
        alt="goals"
        className="placeholder"
        src="../../assets/Landingpage-goals.jpeg" />
    </div>
    <div
      className="col s5">
      <br />
       
      <br />
      <br />
       
      <br />
       
      <br />
      Set goals and subgoals and track your progress. Blacklist sites to limit distractions by blocking or warning you when you have exceeded the time limit you have set.
    </div>
  </div>
  <br />
   
  <br />
  <div
    className="row">
    <div
      className="col s7">
      <br />
       
      <br />
      <br />
       
      <br />
       
      <br />
      Alerts you when you are off task through window notifications using our Chrome extension. Shows prompts to ask you to self-reflect on your progress for the day and how much you have worked towards your goal.
    </div>
    <div
      className="col s5">
      <img
        alt="Chrome extension notification"
        className="placeholder" />
    </div>
  </div>
  <br />
   
  <br />
  <div
    className="row">
    <div
      className="col s7">
      <img
        alt="stats for site"
        className="placeholder"
        src="../../assets/Landingpage-stats.jpeg" />
    </div>
    <div
      className="col s5">
      <br />
       
      <br />
      <br />
       
      <br />
       
      <br />
      Intelligently keeps track of overall data patterns. See how your habits change as you slowly wean yourself off unproductive sites.
    </div>
  </div>
  <br />
   
  <br />
  <div
    className="row">
    <div
      className="col s7">
      <br />
       
      <br />
      <br />
       
      <br />
       
      <br />
      <h5>
        Companion Google Chrome Extension
      </h5>
      Use our Chrome extension to track your tab usage and to send you alert messages and self-reflection questions. Download here.
    </div>
    <div
      className="col s5">
      <img
        alt="Chrome extension icon"
        src="../../assets/icon128.png" />
    </div>
  </div>
  <br />
  <div
    className="center">
    <a
      className="waves-effect waves-light btn-large">
      Get Started
    </a>
  </div>
  <br />
  <h5
    className="center">
     Meet the Team 
  </h5>
  <div
    className="row">
    <div
      className="col">
      <div
        className="white card">
        <div
          className="card-content">
          <span
            className="card-title grey-text text-darken-4">
            Jeff Bentler
          </span>
          <div>
            <img
              alt="Jeff"
              className="circle"
              src="../../assets/jeff_img.jpeg" />
          </div>
        </div>
        <div
          className="card-reveal">
          <span
            className="card-title grey-text text-darken-4">
            Jeff Bentler
            <i
              className="material-icons right">
              close
            </i>
          </span>
        </div>
        <div
          className="card-action">
          <a
            href="https://github.com/bentljm">
            GitHub
          </a>
        </div>
      </div>
    </div>
    <div
      className="col">
      <div
        className="white card">
        <div
          className="card-content">
          <span
            className="card-title grey-text text-darken-4">
            Josephine Chen
          </span>
          <div>
            <img
              alt="Josephine"
              className="circle"
              src="../../assets/josephine_img.jpeg" />
          </div>
        </div>
        <div
          className="card-reveal">
          <span
            className="card-title grey-text text-darken-4">
            Josephine Chen
            <i
              className="material-icons right">
              close
            </i>
          </span>
        </div>
        <div
          className="card-action">
          <a
            href="https://github.com/Josephine-Chen">
            GitHub
          </a>
        </div>
      </div>
    </div>
    <div
      className="col">
      <div
        className="white card">
        <div
          className="card-content">
          <span
            className="card-title grey-text text-darken-4">
            Ian George
          </span>
          <div>
            <img
              alt="Ian"
              className="circle"
              src="../../assets/ian_img.jpeg" />
          </div>
        </div>
        <div
          className="card-reveal">
          <span
            className="card-title grey-text text-darken-4">
            Ian George
            <i
              className="material-icons right">
              close
            </i>
          </span>
        </div>
        <div
          className="card-action">
          <a
            href="https://github.com/OppenheimerAndTheMartians">
            GitHub
          </a>
        </div>
      </div>
    </div>
    <div
      className="col">
      <div
        className="white card">
        <div
          className="card-content">
          <span
            className="card-title grey-text text-darken-4">
            Sherry Hsu
          </span>
          <div>
            <img
              alt="Sherry"
              className="circle"
              src="../../assets/sherry_img.jpeg" />
          </div>
        </div>
        <div
          className="card-reveal">
          <span
            className="card-title grey-text text-darken-4">
            Sherry Hsu
            <i
              className="material-icons right">
              close
            </i>
          </span>
        </div>
        <div
          className="card-action">
          <a
            href="https://github.com/SherryH">
            GitHub
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Settings (Snapshot) Settings renders 1`] = `
<div>
  <h1>
     Settings 
  </h1>
  <h3>
     Blacklist: 
  </h3>
  <table
    className="">
    <thead>
      <tr>
        <th
          data-field="id">
          Site
        </th>
        <th
          data-field="type">
          Type
        </th>
        <th
          data-field="limit">
          Time Limit
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          www.facebook.com
        </td>
        <td>
          Blackout
        </td>
        <td>
          N/A
        </td>
      </tr>
    </tbody>
  </table>
  <br />
  <div
    className="row">
    <div
      className="col input-field s6">
      <input
        className=""
        defaultValue={undefined}
        id="input_6"
        onChange={[Function]}
        placeholder={undefined}
        type="text" />
      <label
        className=""
        data-error={undefined}
        data-success={undefined}
        htmlFor="input_6">
        Add Site
      </label>
    </div>
    <div
      className="col input-field s3">
      <label
        className="active"
        data-error={undefined}
        data-success={undefined}
        htmlFor="input_7">
        Type
      </label>
      <select
        className=""
        defaultValue="1"
        id="input_7"
        multiple={undefined}>
        <option
          value="1">
          Blackout
        </option>
        <option
          value="2">
          Block after exceeding
        </option>
        <option
          value="3">
          Warn after exceeding
        </option>
      </select>
    </div>
    <div
      className="col input-field s3">
      <input
        className=""
        defaultValue={undefined}
        id="input_8"
        onChange={[Function]}
        placeholder={undefined}
        type="text" />
      <label
        className=""
        data-error={undefined}
        data-success={undefined}
        htmlFor="input_8">
        Time Limit (min)
      </label>
    </div>
  </div>
  <h3>
     Personalization: 
  </h3>
  <div
    className="row">
    <div
      className="col input-field s12">
      <input
        className=""
        defaultValue={undefined}
        id="input_9"
        onChange={[Function]}
        placeholder={undefined}
        type="text" />
      <label
        className=""
        data-error={undefined}
        data-success={undefined}
        htmlFor="input_9">
        Image
      </label>
    </div>
  </div>
  <div
    className="row">
    <div
      className="col input-field s12">
      <input
        className=""
        defaultValue={undefined}
        id="input_10"
        onChange={[Function]}
        placeholder={undefined}
        type="text" />
      <label
        className=""
        data-error={undefined}
        data-success={undefined}
        htmlFor="input_10">
        Quote
      </label>
    </div>
  </div>
  <div
    className="row">
    <div
      className="col input-field s2">
      <label
        className="active"
        data-error={undefined}
        data-success={undefined}
        htmlFor="input_11">
        Reminder Type
      </label>
      <select
        className=""
        defaultValue="1"
        id="input_11"
        multiple={undefined}>
        <option
          value="1">
          No Reminder
        </option>
        <option
          value="2">
          Text
        </option>
        <option
          value="3">
          Email
        </option>
      </select>
    </div>
    <div
      className="col input-field s8">
      <input
        className=""
        defaultValue={undefined}
        id="input_12"
        onChange={[Function]}
        placeholder={undefined}
        type="text" />
      <label
        className=""
        data-error={undefined}
        data-success={undefined}
        htmlFor="input_12">
        Number/Email Address
      </label>
    </div>
    <div
      className="col input-field s2">
      <label
        className="active"
        data-error={undefined}
        data-success={undefined}
        htmlFor="input_13">
        Frequency
      </label>
      <select
        className=""
        defaultValue="1"
        id="input_13"
        multiple={undefined}>
        <option
          value="1">
          Daily
        </option>
        <option
          value="2">
          Weekly
        </option>
      </select>
    </div>
  </div>
  <h3>
     Chrome Extension: 
  </h3>
  Forgot to download the extension? Download it here: ___________
</div>
`;

exports[`Subgoal (Snapshot) Subgoal renders 1`] = `
<div>
  <div
    className="row">
    <div
      className="col">
      <input
        className=""
        defaultValue={undefined}
        id="input_5"
        name="group1"
        onChange={[Function]}
        placeholder={undefined}
        type="checkbox"
        value="check" />
      <label
        className=""
        data-error={undefined}
        data-success={undefined}
        htmlFor="input_5">
        Subgoal
      </label>
    </div>
  </div>
</div>
`;
